api/org/commcare/api/persistence/SqliteIndexedStorageUtility.java

```diff
@@ -0,0 +1,460 @@
+/**
+ *
+ */
+package org.commcare.api.persistence;
+
+import org.javarosa.core.services.PrototypeManager;
+import org.javarosa.core.services.storage.EntityFilter;
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+import org.javarosa.core.services.storage.Persistable;
+import org.javarosa.core.services.storage.StorageFullException;
+import org.javarosa.core.util.InvalidIndexException;
+import org.javarosa.core.util.externalizable.DeserializationException;
+import org.sqlite.javax.SQLiteConnectionPoolDataSource;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Vector;
+
+/**
+ * IStorageIndexedUtility implemented on SQLite using JDBC. Contains all the functionality
+ * for interacting with the user's SQLite representation.
+ *
+ * @author wspride
+ */
+public class SqliteIndexedStorageUtility<T extends Persistable> implements IStorageUtilityIndexed<T>, Iterable<T> {
+
+    public static final String DATABASE_FOLDER = "dbs";
+
+    private final Class<T> prototype;
+    private final String tableName;
+    private final String sandboxId;
+
+    public SqliteIndexedStorageUtility(Class<T> prototype, String sandboxId, String tableName) {
+        this.tableName = tableName;
+        this.sandboxId = sandboxId;
+        this.prototype = prototype;
+
+        Connection c = null;
+        try {
+            c = getConnection();
+            SqlHelper.createTable(c, tableName, prototype.newInstance());
+            c.close();
+        } catch (SQLException e) {
+            System.out.println("Couldn't create table: " + tableName + " due to exception: " + e);
+            e.printStackTrace();
+        } catch(InstantiationException | IllegalAccessException | ClassNotFoundException e){
+            throw new RuntimeException(e);
+        } finally{
+            try {
+                if (c != null) {
+                    c.close();
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    Connection getConnection() throws SQLException, ClassNotFoundException {
+
+        File databaseFolder = new File(SqliteIndexedStorageUtility.DATABASE_FOLDER);
+        if (!databaseFolder.exists()) {
+            databaseFolder.mkdir();
```

Comment:
@wpride i think this needs to specify the full path. `mkdir` on `dbs` is probably defaulting `/` on staging. i don't know how you guys handle your settings vars, but we could configure it there, or have touchforms pass it in